app/src/main/java/com/github/se/studentconnect/ui/profile/edit/EditBioViewModel.kt

```
@@ -0,0 +1,94 @@
+package com.github.se.studentconnect.ui.profile.edit
+
+import com.github.se.studentconnect.repository.UserRepository
+import com.github.se.studentconnect.ui.profile.ProfileConstants
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+/**
+ * ViewModel for EditBioScreen.
+ *
+ * Handles bio editing and validation with character limit enforcement and proper error handling.
+ *
+ * @param userRepository Repository for user data operations
+ * @param userId The ID of the user whose bio is being edited
+ */
+class EditBioViewModel(userRepository: UserRepository, userId: String) :
+    BaseEditViewModel(userRepository, userId) {
+
+  // Bio text state
+  private val _bioText = MutableStateFlow("")
+  val bioText: StateFlow<String> = _bioText.asStateFlow()
+
+  // Character count
+  private val _characterCount = MutableStateFlow(0)
+  val characterCount: StateFlow<Int> = _characterCount.asStateFlow()
+
+  // Validation error
+  private val _validationError = MutableStateFlow<String?>(null)
+  val validationError: StateFlow<String?> = _validationError.asStateFlow()
+
+  init {
+    loadCurrentBio()
+  }
+
+  /** Loads the current bio from the repository. */
+  private fun loadCurrentBio() {
+    executeWithErrorHandling(
+        operation = {
+          val user = userRepository.getUserById(userId)
+          val currentBio = user?.bio ?: ""
+          _bioText.value = currentBio
+          _characterCount.value = currentBio.length
+        },
+        onSuccess = { resetState() })
+  }
+
+  /**
+   * Updates the bio text and character count.
+   *
+   * @param newText The new bio text
+   */
+  fun updateBioText(newText: String) {
+    if (newText.length <= ProfileConstants.MAX_BIO_LENGTH) {
+      _bioText.value = newText
+      _characterCount.value = newText.length
+      _validationError.value = null
+    }
```

When `newText` exceeds the limit, consider showing a temporary warning or setting` _validationError` to `ProfileConstants.ERROR_BIO_TOO_LONG` instead of silently ignoring input. This improves UX consistency.